catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -657,4 +571,126 @@ private Filter getTagsQueryFilter() {
                                 .attribute(Metacard.TAGS)
                                 .empty());
     }
+
+    static class QuerySources {
+        private final QueryOperations queryOps;
+
+        private final FrameworkProperties frameworkProperties;
+
+        List<Source> sourcesToQuery = new ArrayList<>();
+
+        Set<ProcessingDetails> exceptions = new HashSet<>();
+
+        boolean addConnectedSources = false;
+
+        boolean addCatalogProvider = false;
+
+        QuerySources(QueryOperations queryOps, FrameworkProperties frameworkProperties) {
```

Comment:
Or, since I assume `QueryOperations` is a singleton, only pass it to the ctor and _not_ to any of the methods? Same for `FrameworkProperties`.